cpp/src/arrow/util/meson.build

```diff
@@ -224,18 +222,25 @@ utility_test_srcs = [
 ]
 
 if host_machine.system() == 'windows'
+    # meson does not natively support manifest files like CMake does
```

Comment:
I don't think the Meson support for manifest files is as nice as CMake's, but I'm also not super familiar with them. Need to research this a bit more...

Reply:
https://github.com/mesonbuild/meson/issues/13041 may be related.

`/MANIFEST:EMBED /MANIFESTINPUT:io_util_test.manifest` linker options may work.
See also: https://learn.microsoft.com/en-us/cpp/build/reference/manifestinput-specify-manifest-input?view=msvc-170